spring/src/main/java/org/wso2/msf4j/spring/MSF4JSpringApplication.java

```diff
@@ -216,20 +216,21 @@ public void setInitializers(List<ApplicationContextInitializer<?>> initializers)
     }
 
     /**
-     * This will add a given service class to the running instnace with given base path.
+     * This will add a given service class to the running instance with given base path.
      *
      * @param configurableApplicationContext ConfigurableApplicationContext of running app
-     * @param serviceClass Service class
-     * @param basePath Base path teh servuce get registered
+     * @param serviceClass                   Service class
+     * @param basePath                       Base path to which the service get registered
      */
-    public void addService(ConfigurableApplicationContext configurableApplicationContext, Class serviceClass,
-                           String basePath) {
+    public SpringMicroservicesRunner addService(ConfigurableApplicationContext configurableApplicationContext,
+                                                Class<?> serviceClass, String basePath) {
```

Comment:
Better if we can add new method rather changing the existing one.

Reply:
This is only a return type change (void -> SpringMicroservicesRunner) and does not break any of the existing implementations.